src/form/Input.js

```diff
@@ -0,0 +1,159 @@
+"use strict";
+(function (root, factory) {
+    if (typeof define === "function" && define.amd) {
+        define(["d3", "../common/HTMLWidget", "css!./Input"], factory);
+    } else {
+        root.form_Form = factory(root.common_HTMLWidget);
+    }
+}(this, function (d3,HTMLWidget) {
+    function Input() {
+        HTMLWidget.call(this);
+        this._class = "form_Input";
+
+        this._tag = "div";
+    }
+    Input.prototype = Object.create(HTMLWidget.prototype);
+    Input.prototype.publish("id", "", "string", "HTML id for the input");
+    Input.prototype.publish("label", "", "string", "Descriptive label");
+    Input.prototype.publish("type", "text", "set", "Input type", ["textbox","number","checkbox","button","select","textarea"]);
+    Input.prototype.publish("validate", null, "string", "Input type");
+    Input.prototype.publish("selectOptions", [], "array", "Array of options used to fill a dropdown list");
+    Input.prototype.publish("value", "", "string", "Input type");
+
+    Input.prototype.testData = function (widget) {
+        return this;
+    };
+    
+    Input.prototype.enter = function (domNode, element) {
+        HTMLWidget.prototype.enter.apply(this, arguments);
+        switch(this.type()){
+            case 'select':
+                this._inputElement = element.append("select");
+                break;
+            case 'textarea':
+                this._inputElement = element.append("textarea");
+                break;
+            case 'button':
+                this._inputElement = element.append("button");
+                break;
+            default:
+                this._inputElement = element.append("input").attr('type',this.type());
+                break;
+        }
+    };
+    Input.prototype.update = function (domNode, element) {
+        HTMLWidget.prototype.update.apply(this, arguments);
+        this._inputElement.attr('id', this.id());
+        
+        switch(this.type()){
+            case 'select':
+                this.checkNodeName('SELECT',element);
```

Comment:
I think its ok to say that "type" can only be set once and has to be set before the first render (we have other examples of this - we should add a flag to the publish property to indicate this?)